### ISSUE TITLE
[BUG] Odd invalid value behaviour when used in combination with the Parametrized Scheduler plugin 

### ISSUE THREAD

**Issue by famod** [jenkinsci/rest-list-parameter-plugin#135]
I have a puzzling issue when combining this plugin with https://plugins.jenkins.io/parameterized-scheduler/.

When I configure a schedule that passes a valid(!) value of a rest-list parameter I'm getting:
```
java.lang.IllegalArgumentException: Illegal value for parameter META_VERSION: 3.0.0.3
   at io.jenkins.plugins.restlistparam.RestListParameterDefinition.checkValue(RestListParameterDefinition.java:242)
   at io.jenkins.plugins.restlistparam.RestListParameterDefinition.createValue(RestListParameterDefinition.java:226)
   at hudson.model.SimpleParameterDefinition.createValue(SimpleParameterDefinition.java:41)
   at org.jenkinsci.plugins.parameterizedscheduler.ParameterizedTimerTrigger.configurePropertyValues(ParameterizedTimerTrigger.java:65)
   at org.jenkinsci.plugins.parameterizedscheduler.ParameterizedTimerTrigger.lambda$checkCronTabsAndRun$0(ParameterizedTimerTrigger.java:85)
   at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
   at org.jenkinsci.plugins.parameterizedscheduler.ParameterizedTimerTrigger.checkCronTabsAndRun(ParameterizedTimerTrigger.java:83)
   at org.jenkinsci.plugins.parameterizedscheduler.Cron.checkTriggers(Cron.java:52)
   at org.jenkinsci.plugins.parameterizedscheduler.Cron.doRun(Cron.java:40)
   at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:92)
   at jenkins.security.ImpersonatingScheduledExecutorService$1.run(ImpersonatingScheduledExecutorService.java:67)
   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
   at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
   at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
   at java.base/java.lang.Thread.run(Thread.java:829)
```
(`3.0.0.3` is a valid value in this example, `META_VERSION` ist the parameter name)

After some trial and error I found out that I have to open the "Build with parameters" page of the job at least once manually to get the value accepted.
It's as if the allowed values are populated lazily and are not there yet when the plugin is invoked like this: https://github.com/jenkinsci/parameterized-scheduler-plugin/blob/master/src/main/java/org/jenkinsci/plugins/parameterizedscheduler/ParameterizedTimerTrigger.java#L63-L65

Does that make any sense?

**Comment by h1dden-da3m0n** [comment]
Interesting, I will try and replicate this.
I know for sure that I have had some errors in my logic when validating if a value is valid or not due to initialisation errors. So I it most likely is my code trying to initialize at the wrong time. I really want to re-implement the entire thing eventually :sweat_smile: 

Thank you for reporting this and your investigation!

**Comment by famod** [comment]
@h1dden-da3m0n have you been able to reproduce this? Do you need more info?

**Comment by h1dden-da3m0n** [comment]
The incremental of the PR: [rest-list-parameter-1.6.2-rc312.f6b_c4f22a_74b_.hpi](https://ci.jenkins.io/job/Plugins/job/rest-list-parameter-plugin/job/PR-136/1/artifact/io/jenkins/plugins/rest-list-parameter/1.6.2-rc312.f6b_c4f22a_74b_/rest-list-parameter-1.6.2-rc312.f6b_c4f22a_74b_.hpi)

Source: [Jenkins-CI Build](https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Frest-list-parameter-plugin/detail/PR-136/1/artifacts)

**Comment by famod** [comment]
@h1dden-da3m0n that was quick, thanks.

I gave it a try and it's looking good now!

Does the fix guarantee that it always has the current values?
By the first look it seems it only kicks in if the list of values is empty? What if the rest endpoint provides different values?

**Comment by h1dden-da3m0n** [comment]
Hey @famod first of all thank you for making sure that quick fix works as expected.

Regarding your second concern, yes, the current PR proposes only ensure that the list gets initialized and thereby fixes part of the behaviour. I did think of simply calling `getValues` every time, but am a tat cautions regarding that due to possible performance implications and further unexpected behaviour.

In theory, everything should be just fine and due to the caching opts the user can configure decently performing, but I cannot say that for sure.
Given the fact that the class life-cycle in Jenkins looks more and more like that the properties of the `ParameterDefinition` class surviving for way longer than I expected when initially concerting the original logic of that class (most of the code is still v1.0.0 code) the PR as it stands may not fix this to 100%.

I will do a few more tests to ensure that simply calling a value update at parameter value creation is fine (with all the implications of request caching) and then update the PR accordingly.

With those changes this should be fixed for good hopefully and result in a class property to drop in the eventual rewrite of my own mess xD

> Note: please excuse my ramble, I sometimes think and write down my thoughts.

**Comment by famod** [comment]
Cool, thanks a lot! I will remind you eventually. 😁

**Comment by famod** [comment]
Fix confirmed with 1.6.2 in my "production" setup. 👍
